perceptia-one/perceptia/src/components/QuizGallery.js

```diff
@@ -52,33 +57,87 @@ class QuizGallery extends React.Component {
 
             })
             .then(list => this.setState({
-                quizlist: list
+                quizlist: list,
+                currentquiz: list
             }))
     }
 
-    // handleChildClick(component, event) {
-    //     this.setState({
-    //         selectedQuiz:component.props.quizName
-    //     })
-    //     this.props.liftStateUp(component.props.quizName)
-    // }
+    fetchStatus(InProgress, quiz) {
+        if (InProgress) {
+            fetch(`${constants.api.url}/api/v1/anyquiz/status/` + quiz)
+                .then(response => response.json())
+                .then(response => this.setState({
+                    fetchStatus: response.status_string
+                }))
+        }
+
+    }
+
+    handleSearchQuiz(quiz) {
+     
+        if (quiz == "") {
+            this.setState({
+                currentquiz: this.state.quizlist
+            })
+        } else if (this.state.quizlist.includes(quiz)) {
+            var temp = [quiz]
+            this.setState({
+                currentquiz: temp
+            })
+        } else {
+            this.setState({
+                fetchInProgress: true
+            })
+
+            fetch(`${constants.api.url}/api/v1/anyquiz/read/` + quiz)
+                .then(response => {
+                    if (!response.ok) {
+                        throw Error(response.statusText)
+                    }
+                    return response
+                })
+                .then(response => {
+                    var temp = [quiz]
+                    this.setState(prevState => ({
+                        fetchInProgress: false, 
+                        quizlist: [...prevState.quizlist, quiz],
+                        currentquiz: temp
+                    }))
+                
+                })
+                .catch(error => this.setState({
+                    fetchInProgress: false,
+                    errorMessage: error
+                }))
+
+            this.interval = setInterval(() => {
+                this.fetchStatus(this.state.fetchInProgress, quiz)
+            }, 5000)
+        }
+
+    }
 
     render() {
         const { classes } = this.props;
         
         return (
             <div>
-            <NavBar/>
+                <NavBar/>
 
-            <Grid container spacing={40}>
-            {this.state.quizlist.map(name => (
-            <Grid item key={name} sm={6} md={4} lg={3}>
-                <QuizInfo quizName={name}/>
-            </Grid>
-            ))}
-            </Grid>
+                <SearchBar onSearchQuiz={this.handleSearchQuiz.bind(this)}/>
+                {
+                    this.state.fetchInProgress ?
+                        <SnackBar text={this.state.fetchStatus} /> : 
```

Comment:
When making a search request (for a quiz that doesn't already exist), there doesn't appear to be any way to exit the "loading" snackbar, as the x button doesn't actually allow you to close the notification. Clearing the search bar doesn't bring back the existing quizzes either (while the search is loading result). Once the quiz does return, it loads the quiz you searched for, but the only way to bring back all the other quizzes is to clear the search text manually and hit enter, which is not intuitive.

Reply:
I agree with @spirrochet, but given that we're a couple days from Capstone, if it would take you more than an hour or two and you can't get to it tomorrow morning so we can get it into the prod. build tomorrow, then we'll merge this as-is.